Documentation/Registry.md

```diff
@@ -0,0 +1,1166 @@
+
+A client SHOULD set the `Accept` header field
+to specify the API version of a request.
+
+```http
+GET /mona/LinkedList/list HTTP/1.1
+Host: packages.example.com
+Accept: application/vnd.swift.registry.v1+json
+```
+
+Valid `Accept` header field values are described by the following rules:
+
+```abnf
+    version     = "1"       ; The API version
+    mediatype   = "json" /  ; JSON (default media type)
+                  "zip"  /  ; Zip archives, used for package releases
+                  "swift"   ; Swift file, used for package manifest
+    accept      = "application/vnd.swift.registry" [".v" version] ["+" mediatype]
+```
+
+A server SHALL set the `Content-Type` and `Content-Version` header fields
+with the corresponding content type and API version number of the response.
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+Content-Version: 1
+```
+
+If a client sends a request without an `Accept` header,
+a server MAY either return `400 Bad Request` or
+process the request using an API version that it chooses,
+making sure to set the `Content-Type` and `Content-Version` headers accordingly.
+
+If a client sends a request with an `Accept` header
+that specifies an unknown or invalid API version,
+a server SHOULD respond with a status code of `400` (Bad Request).
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Version: 1
+Content-Type: application/problem+json
+Content-Language: en
+
+{
+   "detail": "invalid API version"
+}
+```
+
+If a client sends a request with an `Accept` header
+that specifies a valid but unsupported API version,
+a server SHOULD respond with a status code of `415` (Unsupported Media Type).
+
+```http
+HTTP/1.1 415 Unsupported Media Type
+Content-Version: 1
+Content-Type: application/problem+json
+Content-Language: en
+
+{
+   "detail": "unsupported API version"
+}
+```
+
+### 3.6. Package identificaation
+
+A package may declare external packages as dependencies in its manifest.
+Each package dependency may specify a requirement
+on which versions are allowed.
+
+An external package dependency may itself have
+one or more external package dependencies,
+known as <dfn>transitive dependencies</dfn>.
+When multiple packages have dependencies in common,
+Swift Package Manager determines which version of that package should be used
+(if any exist that satisfy all specified requirements)
+in a process called <dfn>package resolution</dfn>.
+
+Each external package is uniquely identified
+by a scoped identifier in the form `scope.package-name`.
+
+#### 3.6.1 Package scope
+
+A *scope* provides a namespace for related packages within a package registry.
+A package scope consists of alphanumeric characters and hyphens.
+Hyphens may not occur at the beginning or end,
+nor consecutively within a scope.
+The maximum length of a package scope is 39 characters.
+A valid package scope matches the following regular expression pattern:
+
+```regexp
+\A[a-zA-Z\d](?:[a-zA-Z\d]|-(?=[a-zA-Z\d])){0,39}\z
```

Comment:
* ASCII `0-9` instead of Unicode `\d` digits?
* Maximum length: 1 + 38 characters?

```suggestion
\A[a-zA-Z0-9](?:[a-zA-Z0-9]|-(?=[a-zA-Z0-9])){0,38}\z
```